internal/declarative/v2/spec.go

```diff
@@ -10,6 +10,12 @@ type SpecResolver interface {
 	Spec(ctx context.Context, object Object, remoteClient client.Client) (*Spec, error)
 }
 
+type RenderMode string
```

Comment:
I believe we can completely drop this `RenderMode` also since currently there is only one render mode anyway, but let's create a follow up issue to address this tech debts.